### PR TITLE
Cynthia's portfolio -- About and Travel Log section

### PR DESCRIPTION
**Pull request by cynthia-doc**
Sorry I forgot to pull request as I commit! This pull request has everything I've been working on so far.

This is my website so far: https://hcynth-step-2020.wl.r.appspot.com
[HTML Preview](https://htmlpreview.github.io/?https://github.com/cynthia-doc/my-portfolio/blob/CynthiaDevelopment/portfolio/src/main/webapp/index.html)

I have a navigation bar linked to each section of the webpage.

The About section is split into two halves and has an intro paragraph and a picture of me:)) 

The Travel Log section has ten scenic photos I took and presented as a slide show, with location of the place at the bottom. I had to treat the first photo as a special case with its own id to ensure that it shows up when the website is first loaded (the default display is "none")

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -12,11 +12,36 @@
...
 
+var slideIndex = 1;
+
+// Next/previous controls
+function plusSlides(n) {
+  showSlides(slideIndex += n);
+}
+
+function showSlides(n) {
+  var i;
```
Comment (Alfredhgoog) [inline portfolio/src/main/webapp/script.js]: You could declare the variable with the loop like for (var i = 0 ; i < slides.length; i ++) {...}

And in general, prefer using let and const over var for scope safety.

https://google.github.io/styleguide/jsguide.html#appendices-legacy-exceptions-var

File: portfolio/src/main/webapp/style.css [hunk 2]
```diff
@@ -1,9 +1,190 @@
+@import url('https://fonts.googleapis.com/css?family=Major Mono Display');
+@import url('https://fonts.googleapis.com/css?family=Codystar');
+@import url('https://fonts.googleapis.com/css?family=Nothing You Could Do');
```
Comment (ricazhang) [inline portfolio/src/main/webapp/style.css]: It is recommended to omit the quotation marks: https://google.github.io/styleguide/htmlcssguide.html#CSS_Quotation_Marks

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (Alfredhgoog) [inline portfolio/src/main/webapp/script.js]: prefer 'index' over 'i'. More descriptive names preferred in general

https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -1,9 +1,190 @@
...
+
 #content {
-  margin-left: auto;
-  margin-right: auto;
-  width: 650px;
+  margin-left: 15%;
+  margin-right: 15%;
```
Comment (ricazhang) [inline portfolio/src/main/webapp/style.css]: There is a shorthand when the left and right padding are the same
https://www.w3schools.com/css/css_padding.asp
https://google.github.io/styleguide/htmlcssguide.html#Shorthand_Properties

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -12,11 +12,36 @@
...
+  var slides = document.getElementsByClassName("mySlides");
+  var firstSlide = document.getElementById("firstSlide");
+  if (n > slides.length + 1) {slideIndex = 1}
+  if (n < 1) {slideIndex = slides.length + 1}
+  for (i = 0; i < slides.length; i++) {
+      slides[i].style.display = "none";
+  }
+  if (slideIndex == 1){
```
Comment (Alfredhgoog) [inline portfolio/src/main/webapp/script.js]: Space before opening bracket

Comment (Alfredhgoog) [inline portfolio/src/main/webapp/script.js]: Be aware of JS type coercion. Use '===' and '!==' instead of '==' and '!=' except when it's desirable.

https://google.github.io/styleguide/jsguide.html#features-equality-checks

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -12,11 +12,36 @@
...
+      firstSlide.style.display = "block";
+  }
+  else{
+      slides[slideIndex - 2].style.display = "block";
```
Comment (Alfredhgoog) [inline portfolio/src/main/webapp/script.js:34]: This seems a bit magical. What happens when slideIndex === 0?

Can you add a comment before the if to explain this logic?

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -1,18 +1,131 @@
...
-      <p>Click here to get a random message:</p>
-      <button onclick="addRandomGreeting()">Hello</button>
-      <div id="greeting-container"></div>
+
+        <h1>Cynthia HU</h1>
+        <ul class="navbar">
+            <li><a href="#About">About</a></li>
+            <li><a href="#TechCorner">Tech Corner</a></li>
+            <li><a href="#TravelLog">Travel Log</a></li>
+            <li><a href="#Contact">Contact</a></li>
```
Comment (ricazhang) [inline portfolio/src/main/webapp/index.html]: The naming convention for CSS classes is usually all lowercase delimited by a hyphen. In this case it would be `#tech-corner` https://google.github.io/styleguide/htmlcssguide.html#Capitalization

File: portfolio/src/main/webapp/style.css [hunk 7]
```diff
@@ -1,9 +1,190 @@
...
+}
+
+#Contact {
+    background-color: #F0EDE5;
+    text-align: right;
+}
+
+
+img.image {
+    float: right;
```
Comment (ricazhang) [inline portfolio/src/main/webapp/style.css]: Flex is kind of like the modern version of float https://css-tricks.com/snippets/css/a-guide-to-flexbox/

File: portfolio/src/main/webapp/style.css [hunk 8]
```diff
@@ -1,9 +1,190 @@
...
+    padding: 20px;
+}
+
+p.intro, div a{
+    width: 50%;
+    white-space: pre-wrap;  
+    word-wrap: break-word;
+
```
Comment (ricazhang) [inline portfolio/src/main/webapp/style.css:43]: Don't forget to use the formatter to format all these files! If the IDE setup is blocking you, you can do formatting in a different PR. Using a formatter would also help automatically fix some of the issues in Alfred's comments!

File: portfolio/src/main/webapp/index.html [hunk 9]
```diff
@@ -1,18 +1,131 @@
...
+            <a target="_blank" rel="noopener noreferrer" href="https://www.google.com/maps/place/Fremont,+CA/data=!4m2!3m1!1s0x808fbf46b7e8caf7:0x8ada313b89d888d4?sa=X&ved=2ahUKEwj4sOqN_NfpAhWX4J4KHU3xDUMQ8gEwIHoECBoQBg">Where I've been spending my life at after the first 12 years...</a><br>
+            <a target="_blank" rel="noopener noreferrer" href="https://www.allure.com/story/aquarius-zodiac-sign-personality-traits">If you are into astrology...</a><br>
+            <a target="_blank" rel="noopener noreferrer" href="https://www.16personalities.com/esfj-personality">If you are into 16 personalities...</a><br>
+            <a target="_blank" rel="noopener noreferrer" href="https://www.youtube.com/watch?v=tWN517PBkGc">Oh, and I dance! Here is a throwback to good old days in highschool...</a><br>
+            
+        </div>
+
+        <p>Click here to get a random message:</p>
+        <button onclick="addRandomGreeting()">Hello</button>
```
Comment (Alfredhgoog) [inline portfolio/src/main/webapp/index.html]: The hello button isn't working because this has not been updated to your new JS function name.

File: portfolio/src/main/webapp/style.css [hunk 10]
```diff
@@ -1,9 +1,190 @@
...
+  width: 100%;
+  text-align: center;
+  text-shadow: 2px 2px #2E4A62;
+}
+
+/* Number text (1/3 etc) */
+.numbertext {
+  color: #EAE6DA;
+  font-size: 18px;
+  font-family: "Lato";
```
Comment (ricazhang) [inline portfolio/src/main/webapp/style.css]: I would suggest adding fallback fonts for all of these. https://css-tricks.com/css-basics-fallback-font-stacks-robust-web-typography/

File: portfolio/src/main/webapp/style.css [hunk 11]
```diff
@@ -1,9 +1,190 @@
...
+    padding: 18px;
 }
+
+a:hover {
+    text-decoration: underline;
+}
+
+/* general setting */
+
+
```
Comment (ricazhang) [inline portfolio/src/main/webapp/style.css]: Remove if unused

File: portfolio/src/main/webapp/style.css [hunk 12]
```diff
@@ -1,9 +1,190 @@
...
+    text-align: right;
+}
+
+
+img.image {
+    float: right;
+    padding: 20px;
+}
+
+p.intro, div a{
```
Comment (Alfredhgoog) [inline portfolio/src/main/webapp/style.css]: ditto - space before opening bracket

File: portfolio/src/main/webapp/style.css [hunk 13]
```diff
@@ -1,9 +1,190 @@
...
+  color: #485167;
+  text-decoration: none;
+  text-align: center;
+  padding: 14px 16px;
+} 
+
+li a:hover {
+    color: #B4B7BA;
+}
+/* navigation bar setting */
```
Comment (ricazhang) [inline portfolio/src/main/webapp/style.css]: It is possible that you would later add some other lists to your page. This styling is specific to the navbar so it would be better to specifically target that. Example: `.navbar` and `.navbar li`

File: portfolio/src/main/webapp/style.css [hunk 14]
```diff
@@ -1,9 +1,190 @@
...
+
+
+/* navigation bar setting */
+ul {
+    list-style-type: none;
+    overflow: hidden;
+    margin: 0;
+    padding: 0;
+    text-align: center;
+    position: sticky;
```
Comment (ricazhang) [inline portfolio/src/main/webapp/style.css:239]: I don't think this is working? The navbar doesn't stick to the top for me.

File: portfolio/src/main/webapp/index.html [hunk 15]
```diff
@@ -1,18 +1,131 @@
...
+
+        <div id="TechCorner">
+            <h2>Tech Corner</h2>
+        </div>
+
+        <div id="TravelLog">
+            <h2>Travel Log</h2>
+            <p>Oh the places I go...</p>
+            <!-- Slideshow container -->
+            <div class="slideshow-container">
```
Comment (Alfredhgoog) [inline portfolio/src/main/webapp/index.html:96]: Great implementation of a flipping photo carousel.

One minor improvement suggested (optional to work on based on your preference): The photos are the last section of the home page and they are not the same height. Therefore while flipping the photos, sometimes the viewport scroll position was forced upwards because of a photo with less height. For good viewing experience the scroll position should remain the same unless user scrolls the page.

You could fix it by simply setting a fixed height for this section.

**Review by Alfredhgoog — COMMENTED** [review]
Thanks for sending the PR. This is a great looking personal site!

File: portfolio/src/main/webapp/script.js [hunk 16]
```diff
@@ -12,11 +12,36 @@
...
+
+// Next/previous controls
+function plusSlides(n) {
+  showSlides(slideIndex += n);
+}
+
+function showSlides(n) {
+  var i;
+  var slides = document.getElementsByClassName("mySlides");
+  var firstSlide = document.getElementById("firstSlide");
```
Comment (ricazhang) [inline portfolio/src/main/webapp/script.js]: Prefer let over var https://google.github.io/styleguide/jsguide.html#features-use-const-and-let

File: portfolio/src/main/webapp/script.js [hunk 17]
```diff
@@ -12,11 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var slideIndex = 1;
+
+// Next/previous controls
```
Comment (ricazhang) [inline portfolio/src/main/webapp/script.js]: jsdoc https://google.github.io/styleguide/jsguide.html#jsdoc-general-form

File: portfolio/src/main/webapp/style.css [hunk 18]
```diff
@@ -1,9 +1,190 @@
...
+
+#Contact {
+    background-color: #F0EDE5;
+    text-align: right;
+}
+
+
+img.image {
+    float: right;
+    padding: 20px;
```
Comment (ricazhang) [inline portfolio/src/main/webapp/style.css]: Generally padding is for spacing inside something and margin is for spacing between elements. The behavior tends to be similar but the semantics is different

File: portfolio/src/main/webapp/index.html [hunk 19]
```diff
@@ -1,18 +1,131 @@
...
+            <img src="/images/profile3.jpg" alt="me in Hokkaido, Japan" 
+                class="image"
+                width="400" height="300"
+                srcset="/images/profile3.jpg 400w, 
+                        /images/profile3.jpg 800w, 
+                        /images/profile3.jpg 1200w"
+                sizes="(max-width: 400px) 200px,
+                        (max-width: 800px) 400px,
+                        (max-width: 1200px) 800px,
+                        1200px"
```
Comment (ricazhang) [inline portfolio/src/main/webapp/index.html]: lmk if I'm testing this wrong but it seems like the image size is the same regardless how wide my browser is

**Review by ricazhang — COMMENTED** [review]
Love the website so far! It looks great :)

File: portfolio/src/main/webapp/style.css (hunk 14 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/style.css:202]: it is sticking when I run it locally but not on the deployed website:// do you perhaps know why it is inconsistent?

Comment (cynthia-doc) [reply portfolio/src/main/webapp/style.css:239]: I didn't really change this portion of the code but somehow it was working in the previous deployed website but now it is:)

File: portfolio/src/main/webapp/style.css (hunk 18 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/style.css]: done (I put image into a image-box div so I can right center the image, so the padding is in image-box now)

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/script.js]: done!

Comment (cynthia-doc) [reply portfolio/src/main/webapp/script.js]: done!

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/script.js]: done

Comment (cynthia-doc) [reply portfolio/src/main/webapp/script.js]: I changed this to <2, which addresses the issue of when slideIndex is 0 (technically would never happen since when the slideIndex is set to number of slides + 1 when slideIndex is less than 1)

File: portfolio/src/main/webapp/index.html (hunk 9 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/index.html]: I changed this section to display favorite movies, and I got rid of the button

File: portfolio/src/main/webapp/index.html (hunk 15 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/index.html:96]: instead of fixing the height, I increased the vertical padding of the buttons so that the user doesn't have to move the mouse to still click the button.

File: portfolio/src/main/webapp/style.css (hunk 2 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/style.css]: done

File: portfolio/src/main/webapp/style.css (hunk 3 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/style.css]: done

File: portfolio/src/main/webapp/index.html (hunk 6 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/index.html]: done, changed the ones not following the guide

File: portfolio/src/main/webapp/style.css (hunk 7 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/style.css]: I decided not to do flex because on smaller websites, I want the text to still wrap around the image, instead of being strictly half and half:)

File: portfolio/src/main/webapp/style.css (hunk 8 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/style.css:43]: will do:)

File: portfolio/src/main/webapp/style.css (hunk 10 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/style.css]: done

File: portfolio/src/main/webapp/style.css (hunk 11 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/style.css]: done

File: portfolio/src/main/webapp/style.css (hunk 13 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/style.css]: done:)

File: portfolio/src/main/webapp/script.js (hunk 16 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/script.js]: I left one variable (slideIndex) as var since I wanted it to be a global variable, but changed the rest to let/const

File: portfolio/src/main/webapp/script.js (hunk 17 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/script.js]: done

File: portfolio/src/main/webapp/index.html (hunk 19 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/index.html]: I didn't really change this portion of the code but somehow it was working in the previous deployed website but now it is:)

File: portfolio/src/main/webapp/script.js (hunk 16 above)
Comment (ricazhang) [reply portfolio/src/main/webapp/script.js]: It looks like slideIndex is in the global scope and using let wouldn't change the fact that it is global. It is still better practice to use let there. Here's a pretty good article on the differences if you're curious: https://tylermcginnis.com/var-let-const/

File: portfolio/src/main/webapp/script.js (hunk 17 above)
Comment (ricazhang) [reply portfolio/src/main/webapp/script.js]: The format is:
`/** the documentation */`

https://google.github.io/styleguide/jsguide.html#jsdoc-general-form

Comment (cynthia-doc) [reply portfolio/src/main/webapp/script.js]: ohh it's not showing up here but I did change the format in the new commits. Ty for the guide tho!

File: portfolio/src/main/webapp/script.js (hunk 16 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/script.js]: I see! I will change it to let:))

File: portfolio/src/main/webapp/script.js [hunk 20]
```diff
@@ -12,23 +12,23 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** slideIndex is a global variable */
```
Comment (Alfredhgoog) [inline portfolio/src/main/webapp/script.js]: This comment is probably not needed because the code explains itself.

File: portfolio/src/main/webapp/script.js [hunk 21]
```diff
@@ -12,23 +12,23 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** slideIndex is a global variable */ 
 var slideIndex = 1;
 
-// Next/previous controls
+/** Next/previous controls */
```
Comment (Alfredhgoog) [inline portfolio/src/main/webapp/script.js]: Nit comment on method comment - "Flips slides to the previous or next page."

https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments
Method descriptions begin with a verb phrase that describes what the method does. This phrase is not an imperative sentence, but instead is written in the third person, as if there is an implied This method ... before it.

Comment (Alfredhgoog) [reply portfolio/src/main/webapp/script.js]: No need to change every occurrence in this PR, but please adopt this practice in future projects :)

**Review by Alfredhgoog — APPROVED** [review]
Approved with minor comments

File: portfolio/src/main/webapp/script.js (hunk 16 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/script.js]: just changed it this morning!

**Review by ricazhang — APPROVED** [review]

File: portfolio/src/main/webapp/index.html [hunk 22]
```diff
@@ -1,18 +1,177 @@
...
+            <li><a href="#about">About</a></li>
+            <li><a href="#tech-corner">Tech Corner</a></li>
+            <li><a href="#travel-log">Travel Log</a></li>
+            <li><a href="#contact">Contact</a></li>
+        </ul>
+        <div id="about">
+            <h2>About</h2>
+           
+            <div class="image-box">
+                <img src="/images/profile3.jpg" alt="me in Hokkaido, Japan" class="image" width="560px" height="420px" align="middle">
```
Comment (ricazhang) [inline portfolio/src/main/webapp/index.html:24]: If you change these to "./images/profile3.jpg" then it will work in the HTML preview.

File: portfolio/src/main/webapp/index.html [hunk 23]
```diff
@@ -1,18 +1,177 @@
...
+        <div id="travel-log" class="section">
+            <h2>Travel Log</h2>
+            <p>Oh the places I go...</p>
+            <!-- Slideshow container -->
+            <div class="slideshow-container"> 
+
+                <!-- Full-width images with number and caption text -->
+                <div id="first-slide" class="fade">
+                    <div class="numbertext">1 / 10</div>
+                    <img src="/images/Shenzhen.jpg" style="width:100%">
```
Comment (ricazhang) [inline portfolio/src/main/webapp/index.html:101]: Same here

File: portfolio/src/main/webapp/index.html [hunk 24]
```diff
@@ -1,18 +1,177 @@
...
+
+                <div class="slides fade">
+                    <div class="numbertext">5 / 10</div>
+                    <img src="/images/Beijing.jpg" style="width:100%">
+                    <div class="text">Beijing, China</div>
+                </div>
+
+                <div class="slides fade">
+                    <div class="numbertext">6 / 10</div>
+                    <img src="/images/Hongkong.jpg" style="width:100%">
```
Comment (ricazhang) [inline portfolio/src/main/webapp/index.html]: For all the images, if you prefix these with a period, ex: "./images/profile3.jpg", then it will work in the HTML preview.

**Review by ricazhang — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/index.html (hunk 24 above)
Comment (cynthia-doc) [reply portfolio/src/main/webapp/index.html]: done!

File: portfolio/src/main/webapp/index.html (hunk 15 above)
Comment (Alfredhgoog) [reply portfolio/src/main/webapp/index.html:96]: Smart solution.

**Review by ricazhang — APPROVED** [review]
